components/Pagination/react.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames/bind'
+import { React as Button } from 'components/Button'
+import { generateVisiblePages } from './generateVisiblePages'
+
+import styles from './style.module.scss'
+
+const cx = classNames.bind(styles)
+
+const Pagination = ({
+  page,
+  totalItems,
+  itemsPerPage,
+  onChange,
+  prevLabel,
+  nextLabel,
+  viewingLabel,
+}) => {
+  // Calculate total pages needed & generate an array of page numbers.
+  const totalPages = Math.ceil(totalItems / itemsPerPage)
+  const pageNumbers = [...Array(totalPages).keys()].map(n => n + 1)
+
+  // Generate the list of visible page numbers.
+  const visiblePageNumbers = generateVisiblePages(page, totalPages)
+
+  // Define our first & last page numbers.
+  const firstPageNumber = 1
+  const lastPageNumber = pageNumbers[pageNumbers.length - 1]
+
+  // Define our previous and next page.
+  const prevPageNumber = page - 1
+  const nextPageNumber = page + 1
+
+  // Calculate the current item range for the current page.
+  const viewingMin = page * itemsPerPage - (itemsPerPage - 1)
+  const viewingMax = page === lastPageNumber ? totalItems : page * itemsPerPage
+  const viewingRange = `${viewingMin}-${viewingMax} of ${totalItems}`
+
+  return (
+    <div className={cx('pagination')}>
+      <div className={cx('pagination__info')}>
+        <span className={cx('pagination__info-viewing-label')}>{viewingLabel}</span>
```

Comment:
I saw in your default prop for `viewingLabel`, you had a space appended to the end. Maybe we should add that spacing with CSS, as that feels like a hidden instruction.

Reply:
@tarikhamilton Agreed. Component styles now handle the trailing space.